### DISCUSSION TITLE
I'm expected to convert html editorState to JSON, but couldn't get it working.

### DISCUSSION THREAD

**Discussion by OttlikG** [facebook/lexical#6352] [General]
### Source code
https://codesandbox.io/p/sandbox/lexical-plain-text-example-forked-7j24ys?file=%2Fsrc%2FEditor.js

I have a `MyLexicalPlugin` where I load a html into the editorState. And I'm expected to get back the Menu in the editor State.

For some reason the only thing I get back is an empty children.

Could you help me get this working and give me a Menu paragraph as the root's children.

### A bit of summary what I want to achieve:
I'm building an inline website editor. So you can inline edit the text. I parse the editor state to HTML send it to ChatGPT for translation, get back the translated text in the right HTML structure. Parse it on the fly (and this is the important bit, without creating another LexicalComposer instance) to JSON and append the JSON to my website structure.

I read the initial hungarien text under
```
{
   children: {Lexical JSON}
}
```

and I want to append to:
```
{
   children: {Lexical JSON},
   languages: {
       eng: {LexicalJSON}
   }
}
```

And then later on the user able to toggle the language states from the header.

Initially I've got to a point where I got the `const nodes = $generateNodesFromDOM(editor, dom);`
and I recursively walked the children. And manually constructed a root and appended the nodes. Unfortunately this structure sometimes crashed the app so I wanted to programatically get the whole JSON information from the editorState. So as you can see I played around with many ways in CodeSandbox. But nothing really worked. I get back empty children state from the parsed html.
Although when I get the `toJSON` from the rendered editor with the hungarien text it works as expected.

**Comment by 2wheeh** [comment]
your sandbox is not found 
```
Error: Error: (404) Not found
Version: 0.0.332
Environment: production
```

**Reply by OttlikG** [reply]
Sorry, the permissions. Now it should work.

**Reply by 2wheeh** [reply]
`exportJSON` call returns an empty array as children.
https://github.com/facebook/lexical/blob/3bd1351e8916693bf932f4108194ba0147a10c55/packages/lexical/src/nodes/LexicalRootNode.ts#L110

It seems `exportJSON` is called in higher abstractions where push nodes to children array:
https://github.com/facebook/lexical/blob/3bd1351e8916693bf932f4108194ba0147a10c55/packages/lexical-clipboard/src/clipboard.ts#L246

what is the goal here though? if you just want to get json of the editorState, try something like this:
```ts
 editor.update(
    () => {
      const parser = new DOMParser();
      const htmlString = `<h4 class="editor-heading-h4 forum_f685f544-module__YJs25G__className" dir="ltr"><span style="color: rgb(255, 255, 255); white-space: pre-wrap;">Menu</span></h4>`;
      const dom = parser.parseFromString(htmlString, "text/html");

      // Once you have the DOM instance it's easy to generate LexicalNodes.
      const nodes = $generateNodesFromDOM(editor, dom);

      // Select the root
      $getRoot().select();

      // Insert them at a selection.
      $insertNodes(nodes);
    },
    {
      onUpdate: () => {
        const json = editor.getEditorState().toJSON();
        console.log(json);
      },
    }
  );
```

**Reply by OttlikG** [reply]
Thanks for your quick reply.
Yeah but at the moment I have the hungarien words and don't want to display the english version. I just will append to the underlying structure without modifying the current editor state.

**Reply by 2wheeh** [reply]
Sorry didn't check updated lines

**Reply by 2wheeh** [reply]
Just a question, don't languages just need text strings for replacement?

**Reply by OttlikG** [reply]
Yeah, this is what I'm experimenting with now. Just grab the text property of the parsable object and translate that.
